store/store.go

```diff
@@ -1208,8 +1202,56 @@ func (s *Store) ListRefresh(installed []*RefreshCandidate, user *auth.UserState)
 		return nil, respToError(resp, "query the store for updates")
 	}
 
-	res := make([]*snap.Info, 0, len(updateData.Payload.Packages))
-	for _, rsnap := range updateData.Payload.Packages {
+	s.extractSuggestedCurrency(resp)
+
+	return updateData.Payload.Packages, nil
+}
+
+func (s *Store) LookupRefresh(installed *RefreshCandidate, user *auth.UserState) (*snap.Info, error) {
+	cur := currentSnap(installed)
+	if cur == nil {
+		return nil, ErrLocalSnap
+	}
+
+	latest, err := s.checkCandidates([]*currentSnapJson{cur}, user)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(latest) != 1 {
+		return nil, ErrSnapNotFound
+	}
+
+	rsnap := latest[0]
+	rrev := snap.R(rsnap.Revision)
+	if rrev == installed.Revision || findRev(rrev, installed.Block) {
+		return nil, &snap.NoUpdateAvailableError{Snap: rsnap.Name}
```

Comment:
That's a good idea; I'll do that in a followup branch (I have one that already touches errors on client and daemon, as this will)